backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -412,6 +417,13 @@ func (a applicationEventLoopRunner_Action) handleDeleteGitOpsDeplEvent(ctx conte
 	}
 
 	if allErrors == nil {
+		if gitopsDepl != nil && isGitOpsDeploymentDeleted(gitopsDepl) {
```

Comment:
The if statement here is actually doing:
`if gitopsDepl != nil && (gitopsDepl == nil || gitopsDepl.DeletionTimestamp != nil)` which seems a bit odd. But it's logically correct, so maybe that's fine :shrug:.

Reply:
Updated it